mp/src/materialsystem/stdshaders/Neo_MotionVision_Shader_Pass2.cpp

```diff
@@ -6,15 +6,16 @@
 // memdbgon must be the last include file in a .cpp file!!!
 #include "tier0/memdbgon.h"
 
-ConVar mat_neo_mv_bw_tint_color_r("mat_neo_mv_bw_tint_color_r", "0.002", FCVAR_CHEAT, "Additional normalized Red color tint for the grayscale MV effect.", true, -1.0f, true, 1.0f);
-ConVar mat_neo_mv_bw_tint_color_g("mat_neo_mv_bw_tint_color_g", "-0.002", FCVAR_CHEAT, "Additional normalized Green color tint for the grayscale MV effect.", true, -1.0f, true, 1.0f);
-ConVar mat_neo_mv_bw_tint_color_b("mat_neo_mv_bw_tint_color_b", "-0.003", FCVAR_CHEAT, "Additional normalized Blue color tint for the grayscale MV effect.", true, -1.0f, true, 1.0f);
+ConVar mat_neo_mv_gradient_blend_offset("mat_neo_mv_gradient_blend_offset", "0.75", FCVAR_CHEAT | FCVAR_ARCHIVE, "Gradient blend offset");
 
 BEGIN_SHADER_FLAGS(Neo_MotionVision_Pass2, "Help for my shader.", SHADER_NOT_EDITABLE)
 
 BEGIN_SHADER_PARAMS
 SHADER_PARAM(MOTIONEFFECT, SHADER_PARAM_TYPE_TEXTURE, "_rt_MotionVision", "")
 SHADER_PARAM(ORIGINAL, SHADER_PARAM_TYPE_TEXTURE, "_rt_MotionVision_Intermediate2", "")
+SHADER_PARAM(MVTEXTURE, SHADER_PARAM_TYPE_TEXTURE, "dev/mvgrad2", "")
+SHADER_PARAM(NOISETEXTURE, SHADER_PARAM_TYPE_TEXTURE, "dev/noise", "")
+SHADER_PARAM(NOISETRANSFORM, SHADER_PARAM_TYPE_VEC3, "Vector(0, 0, 0)", "")
```

Comment:
I'm having a hard time locating where exactly this stuff is parsed in the engine from amidst the C macro hellscape, but are you sure this is accepted syntax here? Some resources [like VDC](https://developer.valvesoftware.com/wiki/Source_SDK_2013:_Your_Second_Shader#Shader_parameter_types) seem to advocate for square brackets syntax here, such as `[0, 0, 0]`. Also here in the engine code: https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/materialsystem/stdshaders/eye_refract.cpp#L20

Reply:
Could be, I had no idea what to put here but since it wasn't failing immediately I haven't touched it since.

Reply:
[It would certainly seem like the square bracket syntax is the way to go](https://github.com/search?q=repo%3AValveSoftware%2Fsource-sdk-2013+SHADER_PARAM_TYPE_VEC3&type=code), unless you have some example code or a reason to expect the `"Vector"` keyword to exist in this context - I doubt it's actually interpreting the string literal argument as C++ code.